agent_memory_server/utils/redis_query.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from typing import Any
+
+from redisvl.query import AggregationQuery, RangeQuery, VectorQuery
+
+
+class RecencyAggregationQuery(AggregationQuery):
+    """AggregationQuery helper for KNN + recency boosting with APPLY/SORTBY and paging.
+
+    Usage:
+      - Build a VectorQuery or RangeQuery (hybrid filter expression allowed)
+      - Call RecencyAggregationQuery.from_vector_query(...)
+      - Chain .load_default_fields().apply_recency(params).sort_by_boosted_desc().paginate(offset, limit)
+    """
+
+    DEFAULT_RETURN_FIELDS = [
+        "id_",
+        "session_id",
+        "user_id",
+        "namespace",
+        "created_at",
+        "last_accessed",
+        "updated_at",
+        "pinned",
+        "access_count",
+        "topics",
+        "entities",
+        "memory_hash",
+        "discrete_memory_extracted",
+        "memory_type",
+        "persisted_at",
+        "extracted_from",
+        "event_date",
+        "text",
+        "__vector_score",
+    ]
+
+    @classmethod
+    def from_vector_query(
+        cls,
+        vq: VectorQuery | RangeQuery,
+        *,
+        filter_expression: Any | None = None,
+    ) -> RecencyAggregationQuery:
+        agg = cls(vq.query)
+        if filter_expression is not None:
+            agg.filter(filter_expression)
+        return agg
+
+    def load_default_fields(self) -> RecencyAggregationQuery:
+        self.load(self.DEFAULT_RETURN_FIELDS)
+        return self
+
+    def apply_recency(
+        self, *, now_ts: int, params: dict[str, Any] | None = None
+    ) -> RecencyAggregationQuery:
+        params = params or {}
+
+        # Support both old and new parameter names for backward compatibility
+        semantic_weight = float(params.get("semantic_weight", params.get("w_sem", 0.8)))
+        recency_weight = float(
+            params.get("recency_weight", params.get("w_recency", 0.2))
+        )
+        freshness_weight = float(params.get("freshness_weight", params.get("wf", 0.6)))
```

Comment:
More support for unreadable variable names - you don't need to support these, all these variables are new and we aren't adding 'wf' etc.